cumulus/parachains/integration-tests/emulated/bridges/bridge-hub-rococo/src/tests/snowbridge.rs

```diff
@@ -199,7 +262,118 @@ fn register_token() {
 		assert_expected_events!(
 			AssetHubRococo,
 			vec![
-				RuntimeEvent::ForeignAssets(pallet_assets::Event::Created { .. }) => {},
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::Success { .. }) => {},
+			]
+		);
+	});
+}
+
+#[test]
+fn reserve_transfer_token() {
+	BridgeHubRococo::fund_accounts(vec![(
+		BridgeHubRococo::sovereign_account_id_of(MultiLocation {
+			parents: 1,
+			interior: X1(Parachain(DEST_PARA_ID)),
+		}),
+		INITIAL_FUND,
+	)]);
+
+	// Fund ethereum sovereign in asset hub
+	AssetHubRococo::fund_accounts(vec![
+		(ETHEREUM_SOVEREIGN.into(), INITIAL_FUND),
+		(ASSETHUB_SOVEREIGN.into(), INITIAL_FUND),
+		(AssetHubRococoReceiver::get(), INITIAL_FUND),
+	]);
+
+	const WETH_AMOUNT: u128 = 1_000_000_000;
+
+	BridgeHubRococo::execute_with(|| {
+		type RuntimeEvent = <BridgeHubRococo as Chain>::RuntimeEvent;
+		type EthereumInboundQueue =
+			<BridgeHubRococo as BridgeHubRococoPallet>::EthereumInboundQueue;
+		let message = VersionedMessage::V1(MessageV1 {
+			chain_id: CHAIN_ID,
+			command: Command::RegisterToken { token: WETH.into() },
+		});
+		let xcm = EthereumInboundQueue::do_convert(message).unwrap();
+		let _ = EthereumInboundQueue::send_xcm(xcm, DEST_PARA_ID.into()).unwrap();
+		let message = VersionedMessage::V1(MessageV1 {
+			chain_id: CHAIN_ID,
+			command: Command::SendToken {
+				token: WETH.into(),
+				destination: Destination::AccountId32 { id: AssetHubRococoReceiver::get().into() },
+				amount: WETH_AMOUNT,
+			},
+		});
+		let xcm = EthereumInboundQueue::do_convert(message).unwrap();
+		let _ = EthereumInboundQueue::send_xcm(xcm, DEST_PARA_ID.into()).unwrap();
+
+		assert_expected_events!(
+			BridgeHubRococo,
+			vec![
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},
+			]
+		);
+	});
+
+	AssetHubRococo::execute_with(|| {
+		type RuntimeEvent = <AssetHubRococo as Chain>::RuntimeEvent;
+		type RuntimeOrigin = <AssetHubRococo as Chain>::RuntimeOrigin;
+
+		assert_expected_events!(
+			AssetHubRococo,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::Success { .. }) => {},
+			]
+		);
+		let fee_asset = MultiAsset {
+			id: Concrete(MultiLocation { parents: 1, interior: Here }),
+			fun: Fungible(1_000_000_000_000), //1 ROC
+		};
```

Comment:
If you remove this fee asset, the call should pass. The fee_asset is the fee for the target chain(Ethereum). We charge 0 fees on the target chain so we make the fee asset WETH and ignore it when converting the xcm. There still will be an additional fee of DOT charged to the user for delivering the message on AssetHub when they call the reserve transfer extrinsic.